tox.ini

```diff
@@ -1,43 +1,64 @@
+pypi_filter = https://raw.githubusercontent.com/sunpy/sunpy/main/.test_package_pins.txt
+# Run the tests in a temporary directory to make sure that we don't import
+# the package from the source tree
+change_dir = .tmp/{envname}
 description =
     run tests
     figure: with figure tests
-    oldestdeps: and with oldest supported dependencies
-    devdeps: and with development versions of matplotlib and astropy
+    oldestdeps: with the oldest supported version of key dependencies
+    devdeps: with the latest developer version of key dependencies
+pass_env =
+    # A variable to tell tests we are on a CI system
+    CI
+    # Custom compiler locations (such as ccache)
+    CC
+    # Location of locales (needed by sphinx on some systems)
+    LOCALE_ARCHIVE
+    # If the user has set a LC override we should follow it
+    LC_ALL
+set_env =
+    MPLBACKEND = agg
+    devdeps: PIP_EXTRA_INDEX_URL = https://pypi.anaconda.org/astropy/simple https://pypi.anaconda.org/scientific-python-nightly-wheels/simple
 deps =
-    pytest-xdist
-    devdeps: astropy>=0.0.dev0
-    devdeps: matplotlib>=0.0.dev0
-    oldestdeps: numpy==1.21.0
-    oldestdeps: astropy==5.0.6
-    oldestdeps: matplotlib==3.5.0
-    figure-!devdeps-!oldestdeps: astropy==5.3.4
-    figure-!devdeps-!oldestdeps: matplotlib==3.8.1
+    # For packages which publish nightly wheels this will pull the latest nightly
+    devdeps: numpy>=0.0.dev0
+    # Packages without nightly wheels will be built from source like this
+    # devdeps: git+https://github.com/ndcube/ndcube
+    oldestdeps: minimum_dependencies
+# The following indicates which extras_require will be installed
 extras =
     all
     tests
-setenv =
-    devdeps: PIP_EXTRA_INDEX_URL = https://pypi.anaconda.org/astropy/simple https://pypi.anaconda.org/scientific-python-nightly-wheels/simple
-    PYTEST_COMMAND = pytest -v -s -raR --pyargs mpl_animators --cov-report=xml --cov=mpl_animators --cov-config={toxinidir}/setup.cfg {toxinidir}/docs
-allowlist_externals = /bin/sh
+commands_pre =
+    oldestdeps: minimum_dependencies mpl_animators --filename requirements-min.txt
+    oldestdeps: pip install -r requirements-min.txt
+    pip freeze --all --no-input
 commands =
-    pip freeze
-    !figure: {env:PYTEST_COMMAND} {posargs}
-    figure: /bin/sh -c "mkdir -p ./figure_test_images; python -c 'import matplotlib as mpl; print(mpl.ft2font.__file__, mpl.ft2font.__freetype_version__, mpl.ft2font.__freetype_build_type__)' > ./figure_test_images/figure_version_info.txt"
-    figure: /bin/sh -c "pip freeze >> ./figure_test_images/figure_version_info.txt"
-    figure: /bin/sh -c "cat ./figure_test_images/figure_version_info.txt"
-    figure: python -c "import mpl_animators.tests.helpers as h; print(h.get_hash_library_name())"
-    figure: {env:PYTEST_COMMAND} --mpl --mpl-generate-summary=html --mpl-baseline-path=https://raw.githubusercontent.com/sunpy/sunpy-figure-tests/mpl-animators-main/figures/{envname}/ {posargs}
```

Comment:
We need this. Most of the tests in this repo are figure tests.

Reply:
then we don't need the pytest call thats replaced this do we?

Reply:
We will need a version of this with things like `figure: --mpl \` in the existing pytest command.